pkg/kubelet/cm/qos_container_manager_linux.go

```diff
@@ -304,8 +328,19 @@ func (m *qosContainerManagerImpl) setMemoryQoS(configs map[v1.PodQOSClass]*Cgrou
 
 func (m *qosContainerManagerImpl) UpdateCgroups() error {
 	m.Lock()
-	defer m.Unlock()
+	ch := make(chan error, 1)
+	m.queue = append(m.queue, ch)
```

Comment:
Why do you have both a queue and a channel?

Reply:
Maybe what makes sense is to delete the lock, make the updateCh have a much larger buffer, and send the errorCh over that channel?

Reply:
UpdateCgroups should be blocking and should return an error code so for each call we need a channel to signal when the actual update cgroups operation we are waiting for has finished and return the associated error code.